From 176ae963651e47d8bef682b7e37c183f4df6ca0d Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Wed, 20 May 2026 20:33:14 +0200
Subject: [PATCH 2/2] fw/b: Suggestion Popup: Tint properly

* Instead of going by the "isLightTheme" attribute of an app, which
  might not even be set or default to true, go by the configuration
* That way, the text is now properly tinted (light on dark or vice
  versa) - it already follows the system color scheme

Test: Open various apps with edit fields, type something, open
suggestions

Change-Id: I9b5aff57fb7c0a8127ddedc636bb8a3519002a5a
---
 core/java/android/widget/Editor.java | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/core/java/android/widget/Editor.java b/core/java/android/widget/Editor.java
index 06b0c1a4e8ee..88eb3392a532 100644
--- a/core/java/android/widget/Editor.java
+++ b/core/java/android/widget/Editor.java
@@ -4211,12 +4211,11 @@ public class Editor {
         }
 
         private Context applyDefaultTheme(Context originalContext) {
-            TypedArray a = originalContext.obtainStyledAttributes(
-                    new int[]{com.android.internal.R.attr.isLightTheme});
-            boolean isLightTheme = a.getBoolean(0, true);
+            int nightModeFlags = originalContext.getResources()
+                    .getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
+            boolean isLightTheme = nightModeFlags != Configuration.UI_MODE_NIGHT_YES;
             int themeId = isLightTheme ? R.style.ThemeOverlay_Material_Light
                     : R.style.ThemeOverlay_Material_Dark;
-            a.recycle();
             return new ContextThemeWrapper(originalContext, themeId);
         }
 
-- 
2.54.0

